src/resolver/findAllExportedComponentDefinitions.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+import isExportsOrModuleAssignment from '../utils/isExportsOrModuleAssignment';
+import isReactComponentClass from '../utils/isReactComponentClass';
+import isReactCreateClassCall from '../utils/isReactCreateClassCall';
+import isStatelessComponent from '../utils/isStatelessComponent';
+import normalizeClassDefinition from '../utils/normalizeClassDefinition';
+import resolveExportDeclaration from '../utils/resolveExportDeclaration';
+import resolveToValue from '../utils/resolveToValue';
+
+function ignore() {
+  return false;
+}
+
+function isComponentDefinition(path) {
+  return isReactCreateClassCall(path) || isReactComponentClass(path) || isStatelessComponent(path);
+}
+
+function resolveDefinition(definition, types) {
+  if (isReactCreateClassCall(definition)) {
+    // return argument
+    var resolvedPath = resolveToValue(definition.get('arguments', 0));
+    if (types.ObjectExpression.check(resolvedPath.node)) {
+      return resolvedPath;
+    }
+  } else if(isReactComponentClass(definition)) {
+    normalizeClassDefinition(definition);
+    return definition;
+  } else if (isStatelessComponent(definition)) {
+    return definition;
+  }
+  return null;
+}
+
+/**
+ * Given an AST, this function tries to find the exported component definitions.
+ *
+ * The component definitions are either the ObjectExpression passed to
+ * `React.createClass` or a `class` definition extending `React.Component` or
+ * having a `render()` method.
+ *
+ * If a definition is part of the following statements, it is considered to be
+ * exported:
+ *
+ * modules.exports = Definition;
+ * exports.foo = Definition;
+ * export default Definition;
+ * export var Definition = ...;
+ */
+export default function findExportedComponentDefinitions(
+  ast: ASTNode,
+  recast: Object
+): ?NodePath {
```

Comment:
I tried changing it yesterday and got different errors. I'll look into it when I have a bit more time.